### PR TITLE
Add ColPali as reranker

### PR DESCRIPTION
**Pull request by ravi03071991**
# Description

PR to add ColPali as reranker.

Fixes # (issue)

## New Package?

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [ ] Yes
- [ ] No

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [ ] Yes
- [ ] No

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Your pull-request will likely not be merged unless it is covered by some form of impactful unit testing.

- [ ] I added new unit tests to cover this change
- [ ] I believe this change is already covered by existing unit tests

## Suggested Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

**Comment by ravi03071991** [comment]
<p>Yup. It does not work right. Because we cannot use postprocessor with retrievers.</p>

---

 <a href='https://app.reviewnb.com/run-llama/llama_index/pull/16829/discussion/#comment-2457515454'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

File: llama-index-integrations/postprocessor/llama-index-postprocessor-colpali-rerank/tests/test_postprocessor_colpali_rerank.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+def test_class():
+    names_of_base_classes = [b.__name__ for b in ColPaliRerank.__mro__]
+    assert BaseNodePostprocessor.__name__ in names_of_base_classes
+
+
+def test_init():
+    m = ColPaliRerank(top_n=10)
+
+    assert m.model == "vidore/colpali"
```
[the remark this answers is not in the document]
Comment (ravi03071991) [reply llama-index-integrations/postprocessor/llama-index-postprocessor-colpali-rerank/tests/test_postprocessor_colpali_rerank.py:13]: probably `vidore/colpali-v1.2` ?

[the remark this answers is not in the document]
Comment (nerdai) [reply llama-index-integrations/postprocessor/llama-index-postprocessor-colpali-rerank/tests/test_postprocessor_colpali_rerank.py:13]: yeah i think that's a good one

[the remark this answers is not in the document]
Comment (ravi03071991) [reply llama-index-integrations/postprocessor/llama-index-postprocessor-colpali-rerank/tests/test_postprocessor_colpali_rerank.py:13]: Cool. Will update before making a merge.

**Pull request merged**